maven_iuvs/echelle.py

```diff
@@ -0,0 +1,1292 @@
+    # Make sure it's a set (unique values only)
+    all_bad_inds = sorted(list(set([*nan_light_inds, *bad_light_inds, *bad_dark_inds])))
+    dark_subtracted = np.delete(dark_subtracted, all_bad_inds, axis=0)
+
+    d = 0 # Counter for the dark_subtracted array, since we may have modified its shape.
+    # Begin subtracting - don't do it if a dark contains NaNs though
+    if 0 not in all_bad_inds:
+        dark_subtracted[d, :, :] = light_data[0] - first_dark
+        d += 1
+
+    if not np.isnan(second_dark).any():
+        for i in range(1, light_data.shape[0]):
+            if i not in all_bad_inds:
+                dark_subtracted[d, :, :] = light_data[i] - second_dark
+                d += 1
+
+    return dark_subtracted, nan_light_inds, bad_light_inds, bad_dark_inds
+
+
+def get_dark_frames(dark_fits, flbl=None):
+    """
+    Given a fits file containing dark integrations, this will identify and return
+    the first and second dark frames. If more than 2 dark integrations exist,
+    the 2nd through nth dark frame will be averaged to create the second dark.
+    If any resulting dark contains nans, it will be set to None. 
+
+    Parameters
+    ----------
+    dark_fits : IUVSfits or fits file
+                fits file containing dark integraitons.
+
+    Returns
+    ----------
+    first_dark, second_dark: Arrays or None
+                             Dark frames contained within the observation.
+                             None if the recovered frame contains NaN.
+
+    """
+    n_ints_dark = get_n_int(dark_fits)
+
+    if n_ints_dark <= 1: 
+        raise ValueError(f"Error: There are only {n_ints_dark} dark integrations in file {dark_fits.basename}")
+
+    if n_ints_dark == 2:
+        first_dark = dark_fits['Primary'].data[0]
+        if (dark_fits["Observation"].data["ORBIT_NUMBER"] >= 9070) and (flbl == 'disk_i'):  # account for corrupt darks in Corona_13 indisks: adapted from Majd's code.
+            second_dark = first_dark
+        else:
+            # Separate the first and second dark integrations (they have different noise patterns)
+            second_dark = dark_fits['Primary'].data[1]
+    else:
+        first_dark = dark_fits['Primary'].data[0]
+        # If more than 2 additional darks, get the element-wise mean to use as second dark. Ignore nans.
+        second_dark = np.nanmean(np.array([d for d in dark_fits['Primary'].data[1:]]), axis=0) 
+
+    return first_dark, second_dark
+
+
+def pair_lights_and_darks(selected_l1a, dark_idx, verbose=False):
+    """
+    Fills a dictionary, lights_and_darks, with light and dark metadata for a given light observation file,    
+    which makes it easier to process quicklooks. Calls on find_dark_options, so any errors in dark association
+    should be fixed within that function.
+    
+    Parameters
+    ----------
+    selected_l1a : list of dictionaries
+                   selected l1a metadata entries to use
```

Comment:
(presumably a list of light files)

Reply:
Clarified this docstring

Reply:
Fixed by d81b483549da109ff45b25607960cdaf97a7345b